server/schema/resolvers.ts

```diff
@@ -1,18 +1,64 @@
-import { GraphQLDateTime } from 'graphql-iso-date'
-import { chats, messages } from '../db'
+import { GraphQLDateTime } from "graphql-iso-date";
+import { chats, messages } from "../db";
 
 const resolvers = {
-    Date: GraphQLDateTime,
-    Chat: {
-        lastMessage(chat: any){
-            return messages.find( m => m.id === chat.lastMessage)
-        },
+  Date: GraphQLDateTime,
+  Chat: {
+    messages(chat: any) {
+      return messages.filter(m => chat.messages.includes(m.id));
     },
-    Query: {
-        chats() {
-            return chats;
-        },
+    lastMessage(chat: any) {
+      const lastMessage = chat.messages[chat.messages.length - 1];
+      return messages.find(m => m.id === lastMessage);
     },
+  },
+  Query: {
+    chats() {
+      return chats;
+    },
+
+    chat(root: any, { chatId }: any) {
+      return chats.find(c => c.id === chatId);
+    }
+  },
+
+  Mutation: {
+    addMessage(root: any, { chatId, content }: any) {
+      //find index of chat by chatId
+      const chatIndex = chats.findIndex(c => c.id === chatId);
```

Comment:
Array.filter is less ackward to work with

Reply:
-1. Array.filter has a totally different use case. This code is coming from the whatsapp tutorial and it's fine IMO